pkg/annotations/ingress_annotations_test.go

```diff
@@ -215,6 +220,22 @@ var _ = Describe("Test ingress annotation functions", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(actual).To(Equal(true))
 		})
+
+		It("returns true with correct annotation", func() {
+			ing.Annotations[IngressClassKey] = "agis5"
+			ApplicationGatewayIngressClass = "agis5"
+			actual, err := IsApplicationGatewayIngress(ing)
```

Comment:
I would suggest to update IsApplicationGatewayIngress to use DefaultApplicationGatewayIngressClass directly, benefit is you don't have to hack things outside to make the test work

Reply:
I will look into this. I may have follow-up questions.

Reply:
@3quanfeng can you share some additional thoughts in terms of hacking things? Which part would we be trying to avoid? The BeforeEach() block?

Reply:
Inside IsApplicationGatewayIngress, if you use DefaultIngressClass directly, then you don't need to care about ApplicationGatewayIngressClass in your test

Reply:
You mean like this? If yes, how would the ingress class override occur if not in this function?

```go

func IsApplicationGatewayIngress(ing *v1beta1.Ingress) (bool, error) {
	controllerName, err := parseString(ing, IngressClassKey)
	return controllerName == DefaultIngressClass, err
}
```

Reply:
Correct, the change honors the signature as well.

Reply:
This would negate the effect of this change:

https://github.com/garvinmsft/application-gateway-kubernetes-ingress/blob/92d9ac3a915f94e24a7f90883a99250164cd33c2/cmd/appgw-ingress/utils.go#L111

Reply:
The new logic intends to override the default. So that IsApplicationGatewayIngress returns true for the default or any arbitrary value set in IsApplicationGatewayIngress if overriden.
https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/proposals/multiple-gateways-single-cluster.md

Reply:
I see, in this case, we could just set the default("azure/application-gateway") in environment.go.
then we don't need to make ANY changes in ingress_annotations.go, how does it sound?

Reply:
you can set the default in env.Consolidate()

Reply:
I think what you are getting at is to use 1 variable "ApplicationGatewayIngressClass" which contains the default but can be re-assigned in util. In that case, the tests won't cover the scenario where the ingress resource has a class different from the default but matches the new config value. 

What you are suggesting with env.Consolidate() can work but we would need to send env all the way to the handler and not rely on the package variable.